test/graphql/queries/system_query_test.rb

```diff
@@ -935,27 +954,31 @@ class SystemQueryTest < ActiveSupport::TestCase
 
       asc_result = Schema.execute(
         query,
-        variables: { policyId: @profile1.id, sortBy: ['rulesFailed'] },
+        variables: {
+          policyId: @profile1.id,
+          sortBy: ['rulesFailed'],
+          filter: "with_results_for_policy_id = #{@profile1.id}"
+        },
         context: { current_user: @user }
-      )['data']['systems']['edges']
+      )['data']['systems']['edges'].map do |row|
+        row['node']['testResultProfiles'].first['rulesFailed']
+      end
 
       desc_result = Schema.execute(
         query,
-        variables: { policyId: @profile1.id, sortBy: ['rulesFailed:DESC'] },
+        variables: {
+          policyId: @profile1.id,
+          sortBy: ['rulesFailed:DESC'],
+          filter: "with_results_for_policy_id = #{@profile1.id}"
+        },
         context: { current_user: @user }
-      )['data']['systems']['edges']
-
-      returned_profile1 = asc_result.dig(0, 'node', 'testResultProfiles')
-      returned_profile2 = asc_result.dig(1, 'node', 'testResultProfiles')
-
-      returned_profile3 = desc_result.dig(0, 'node', 'testResultProfiles')
-      returned_profile4 = desc_result.dig(1, 'node', 'testResultProfiles')
+      )['data']['systems']['edges'].map do |row|
+        row['node']['testResultProfiles'].first['rulesFailed']
+      end
 
-      assert_equal 1, returned_profile1.dig(0, 'rulesFailed')
-      assert_equal 0, returned_profile2.dig(0, 'rulesFailed')
+      assert_equal asc_result, [0, 1, 2]
```

Comment:
The tests look good but we should probably test a system with `nil` failed rules right?

Reply:
Well, feel free to adjust the tests

Reply:
Yep, working on it! And then I think everything looks good 👍